vite.config.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable camelcase, @typescript-eslint/no-magic-numbers */
+// eslint-env node
+import { readFileSync } from 'fs';
+
+import { defineConfig, type UserConfig } from 'vitest/config';
+import { type ManifestOptions, VitePWA as vitePWA } from 'vite-plugin-pwa';
+import htmlMinifier from 'vite-plugin-html-minifier';
+
+import { externalResources, internalResources } from './src/service-worker';
+
+const manifest: Partial<ManifestOptions> = JSON.parse(readFileSync('./src/manifest.json', { encoding: 'utf8' }));
+
+export default defineConfig(({ mode }) => {
+	const baseUrl = mode === 'production' ? 'https://brand.torontojs.com/' : 'https://localhost:3000/';
+
+	const sslOptions = mode === 'production'
+		? false
+		: {
```

Comment:
I dunno if it's because I'm on a M2 and I know nothing of mkcert other than having to generate them at work also. For somereason my computer insisted on creating .pem files, regardless by changing these lines to: 

```
 {
          cert: readFileSync("./certutil.pem"),
          key: readFileSync("./certutil-key.pem"),
        }
```

... I was able to get it running!